src/algebra/direct_sum/internal.lean

```diff
@@ -178,6 +178,21 @@ instance nat_power_graded_monoid
 { one_mem := by { rw [←one_le, pow_zero], exact le_rfl },
   mul_mem := λ i j p q hp hq, by { rw pow_add, exact submodule.mul_mem_mul hp hq } }
 
+section homogeneous_element
+
+lemma set_like.is_homogeneous_zero [add_monoid ι]
```

Comment:
Maybe `set_like` should be removed, because this is in `submodule` namespace

Reply:
Yes, i think `set_like.is_homogeneous_submodule_zero` would be a better name than the current `submodule.set_like.is_homogeneous_zero`. You can do that with
```suggestion
lemma set_like.is_homogeneous_zero [add_monoid ι]
```
```suggestion
lemma _root_.set_like.is_homogeneous_submodule_zero [add_monoid ι]
```